pym/bob/cmds/jenkins.py

```diff
@@ -628,6 +631,10 @@ def dumpXML(self, orig, nodes, windows, credentials, clean, options, date, autht
                 xml.etree.ElementTree.SubElement(
                     cp, "doNotFingerprintArtifacts").text = "true"
 
+            if archiveSharing:
+                prepareCmds.append("\n# download deps\n# =============")
+                prepareCmds.append(self.__archive.download(d, JenkinsJob._buildIdName(d), JenkinsJob._tgzName(d)))
```

Comment:
This is not indented correctly (not part of the loop above). It will download just the last dependency. Likewise, the download should not be done for shared artifacts if the condition does not match.